federation/pkg/federation-controller/util/cluster_util.go

```diff
@@ -146,3 +149,55 @@ func GetClientsetForCluster(cluster *federation_v1beta1.Cluster) (*fedclientset.
 	}
 	return nil, err
 }
+
+// Returns a bool to indicate if the object should be forwarded to a cluster
+func SendToCluster(cluster *federation_v1beta1.Cluster, objMeta metav1.ObjectMeta) bool {
+	forwardObject := true
+	// Check if there is an annotation for ClusterSelector
+	if val, ok := objMeta.Annotations[federation_v1beta1.FederationClusterSelector]; ok {
+		// Check if the Annotation contains valid data
+		selector := labels.NewSelector()
+		requirements := make([]federation_v1beta1.ClusterSelectorRequirement, 0)
+		if err := json.Unmarshal([]byte(val), &requirements); err == nil {
+			for _, requirement := range requirements {
+				r, err := labels.NewRequirement(requirement.Key, ConvertOperator(requirement.Operator), requirement.Values)
+				if err != nil {
+					glog.V(8).Infof("Unable to convert ClusterSelector Annotation to Requirement: %+v, %s", requirement, err.Error())
+				}
+				selector = selector.Add(*r)
+			}
+			if !selector.Matches(labels.Set(cluster.Labels)) {
+				glog.V(8).Infof("Object %s Selector: %+v does not match Cluster %s labels: %+v", objMeta.Name, selector.String(), cluster.ObjectMeta.Name, cluster.Labels)
+				forwardObject = false
+			}
+		} else {
+			glog.V(8).Infof("Unable to parse ClusterSelector Annotation: %s", err.Error())
```

Comment:
Errors are now returned to the controller to be handled and or returned from there.